Implemented interface MyLinkedListInterface in class MyLinkedList

DanaKhromenko opened the pull request.

DanaKhromenko commented:
Changed the implementation of the loop in getNode() method.

RomanDubovskyi commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -3,45 +3,179 @@
     public boolean remove(T object) {
+        Node<T> node = head;
+        while (node != null) {
+            if ((node.value == null && object == null) || node.value.equals(object)) {
```
```suggestion
            if ((node.value == object) || node.value.equals(object)) {
```
другу частину після ||  можна ще покращити. Наразі там може бути NPE якщо node.value == null і object != null. Тоді перша частина дасть false а друга NPE бо на налі викличемо .equals()

RomanDubovskyi commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -3,45 +3,179 @@
     @Override
     public T set(T value, int index) {
+        Node<T> currentNode = getNode(index);
+        if (currentNode != null) {
```
логіка побудована так, що у нас currentNode не може бути null. Тому можна прибрати цю перевірку і ліст не має ламатись

RomanDubovskyi commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -3,45 +3,179 @@
+                    + index + ", last index = " + (size - 1));
+        }
+        Node<T> currentNode = getNode(index);
+        if (currentNode != null) {
```
аналогічно тут з налом

RomanDubovskyi commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -3,45 +3,179 @@
+        Node<T> currentNode = (moveFromHead ? head : tail);
+        if (moveFromHead) {
+            for (int i = 0; i < index; i++) {
+                if (currentNode == null) {
+                    return null;
+                }
+                currentNode = currentNode.next;
```
```suggestion
                currentNode = currentNode.next;
```

RomanDubovskyi commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -3,45 +3,179 @@
 
     @Override
     public T get(int index) {
+        if (index < 0 || index >= size) {
```
у тебе ця перевірка дублюється у getNode() методі

DanaKhromenko commented:
- replaced statement "(node.value == null && object == null) || node.value.equals(object)" with statement "node.value == object || node.value.equals(object)";
- replaced statements "if (currentNode != null)...";
- replaced doubled index validation in get() method.

RomanDubovskyi commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -3,45 +3,166 @@
     public boolean remove(T object) {
+        Node<T> node = head;
+        while (node != null) {
+            if (node.value == object || node.value.equals(object)) {
```
```suggestion
            if (node.value == object || node.value != null && node.value.equals(object)) {
```
так буде сейв від NPE
